components/org.wso2.carbon.deployment.engine/src/main/java/org/wso2/carbon/deployment/engine/config/DeploymentConfiguration.java

```diff
@@ -40,10 +40,7 @@
     private DeploymentNotifierConfig deploymentNotifier = new DeploymentNotifierConfig();
 
     public DeploymentConfiguration() {
-        repositoryLocation = "${carbon.home}/deployment/";
-        if (Utils.getSystemVariableValue("carbon.home", null) != null) {
-            repositoryLocation = Utils.substituteVariables(repositoryLocation);
-        }
+        repositoryLocation = ConfigurationUtils.substituteVariables("${carbon.home}/deployment/");
```

Comment:
We need to monitor all the deployment folders, i.e., the Server deployment folder and the current runtime's deployment folder.